src/libmongoc/tests/test-mongoc-transactions.c

```diff
@@ -1166,6 +1166,39 @@ test_max_commit_time_ms_is_reset (void *ctx)
    mock_rs_destroy (rs);
 }
 
+void
+test_snapshot_session_prose_1 (void *ctx)
+{
+   mongoc_client_t *client = NULL;
+   mongoc_session_opt_t *session_opts = NULL;
+   mongoc_uri_t *uri = NULL;
+   bson_error_t error;
+   bool r;
+
+   uri = test_framework_get_uri ();
+   client = test_framework_client_new_from_uri (uri, NULL);
```

Comment:
The `test_framework_client_new_from_uri` will not set SSL options that are required when running against a server with SSL enabled. Other tests that use `test_framework_client_new_from_uri` also call `test_framework_set_ssl_opts` right after. 

Since you do not need to construct a URI in this test, you can simplify this to:

```suggestion
   client = test_framework_new_default_client ();
```

That also sets the SSL options.